graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlServiceBuilder.java

```diff
@@ -83,7 +94,51 @@ public GraphqlServiceBuilder schemaFile(File... schemaFiles) {
      * If not set, the {@code schema.graphql} or {@code schema.graphqls} will be imported from the resource.
      */
     public GraphqlServiceBuilder schemaFile(Iterable<? extends File> schemaFiles) {
-        this.schemaFiles.addAll(requireNonNull(schemaFiles, "schemaFiles"));
+        requireNonNull(schemaFiles, "schemaFiles");
+        return schemaUrls0(Streams.stream(schemaFiles)
+                                  .map(file -> {
+                                      try {
+                                          return file.toURI().toURL();
+                                      } catch (MalformedURLException e) {
+                                          throw new UncheckedIOException(e);
+                                      }
+                                  }).collect(toImmutableList()));
+    }
+
+    /**
+     * Adds the schema {@code schemaUrl}s.
+     * If not set, the {@code schema.graphql} or {@code schema.graphqls} will be imported from the resource.
+     */
+    public GraphqlServiceBuilder schemaUrls(String... schemaUrls) {
+        return schemaUrls(ImmutableList.copyOf(requireNonNull(schemaUrls, "schemaUrls")));
+    }
+
+    /**
+     * Adds the schema {@code schemaUrl}s.
+     * If not set, the {@code schema.graphql} or {@code schema.graphqls} will be imported from the resource.
+     */
+    public GraphqlServiceBuilder schemaUrls(Iterable<String> schemaUrls) {
+        requireNonNull(schemaUrls, "schemaUrls");
+        return schemaUrls0(Streams.stream(schemaUrls)
+                                  .map(url -> {
+                                      try {
+                                          return ResourceUtil.getURL(url);
+                                      } catch (FileNotFoundException e) {
+                                          throw new IllegalStateException("Not found schema file(s)");
```

Comment:
How about adding the exception?
```suggestion
                                          throw new IllegalStateException("Not found schema file(s)", e);
```